eng/tools/generator/cmd/v2/common/cmdProcessor.go

```diff
@@ -19,11 +19,48 @@ import (
 func ExecuteGoGenerate(path string) error {
 	cmd := exec.Command("go", "generate")
 	cmd.Dir = path
-	output, err := cmd.CombinedOutput()
-	log.Printf("Result of `go generate` execution: \n%s", string(output))
+
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+
+	stderrPipe, err := cmd.StderrPipe()
 	if err != nil {
-		return fmt.Errorf("failed to execute `go generate` '%s': %+v", string(output), err)
+		return err
+	}
+
+	if err = cmd.Start(); err != nil {
+		return err
+	}
+
+	var stdoutBuffer bytes.Buffer
+	if _, err = io.Copy(&stdoutBuffer, stdoutPipe); err != nil {
+		return err
+	}
+
+	var stderrBuffer bytes.Buffer
+	if _, err = io.Copy(&stderrBuffer, stderrPipe); err != nil {
+		return err
+	}
+
+	fmt.Println(stdoutBuffer.String())
+	if stdoutBuffer.Len() > 0 {
```

Comment:
Can you move this error check section to the place after `cmd.Wait()` as it might miss errors that occur during the commands' execution?